src/afl-fuzz-init.c

```diff
@@ -53,25 +50,60 @@ void bind_to_free_cpu(afl_state_t *afl) {
 
   }
 
+  u8  cpu_used[4096] = {0}, lockfile[PATH_MAX] = "";
+  u32 i;
+
+  if (afl->cpu_to_bind != -1) {
+
+    i = afl->cpu_to_bind;
```

Comment:
The goto spaghetti code will go back to `try` if the current cpu cannot be bound and continue with the next one or crash or something.
set_cpu should instead return if it worked, the normal code may then loop, `-b` should FATAL if the core cannot be bound.

Reply:
Also metaquestion: Why not make this a env var?

Reply:
because this is a PR to google afl and we would loose compatibility if it is not the same command line option.

the goto was there before and I used it to make the code simpler. if the bind fails but cpu_to_bind is != -1 then there is the FATAL. I introduced the feature with as little lines of code changed as possible.
your change might have broken it, will have a look.

yes set_cpu() would be nicer, but I hadn't had the time.

Reply:
also the code is expected to try the next cpu if a bind fails (except for -b, then it FATALs)